libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/client/RetrofitApiClient.java

```diff
@@ -0,0 +1,56 @@
+package net.cyclestreets.api.client;
+
+import net.cyclestreets.api.POI;
+import net.cyclestreets.api.client.dto.FeatureCollection;
+
+import java.io.IOException;
+import java.util.List;
+
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RetrofitApiClient {
+
+  private final V1Api v1Api;
+  private final V2Api v2Api;
+
+  public RetrofitApiClient(String v1Host, String v2Host, String apiKey) {
```

Comment:
Actually, I'd go one further and say that having a constructor with three `String` parameters is a bad smell.  Consequently, I've changed this to use a builder... see what you think.

Regarding updating http references to https, are you referring just to the `ApiClient` itself (https://github.com/cyclestreets/android/blob/master/libraries/cyclestreets-core/src/main/java/net/cyclestreets/api/ApiClient.java#L65) or more widely?  If the latter, I think it warrants its own PR separate from this one.